src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -970,6 +987,12 @@ private void CheckAllowedProjectChanges(ProjectChanges projectChanges)
                 throw new NotSupportedException(WorkspacesResources.ChangingCompilationOptionsNotSupported);
             }
 
+            if (projectChanges.OldProject.EmitOptions != projectChanges.NewProject.EmitOptions
+                && !this.CanApplyChange(ApplyChangesKind.ChangeEmitOptions))
```

Comment:
is != the right operator to use for emit options?  Does it implement those value equality operators? Should we be using .Equals?